tiledb/sm/rest/rest_client_remote.cc

```diff
@@ -234,13 +234,14 @@ shared_ptr<ArraySchema> RestClientRemote::post_array_schema_from_rest(
     bool include_enumerations) {
   serialization::LoadArraySchemaRequest req(include_enumerations);
 
-  Buffer buf;
+  SerializationBuffer buff{serialization_buffer_resource()};
   serialization::serialize_load_array_schema_request(
-      config, req, serialization_type_, buf);
+      config, req, serialization_type_, buff);
 
-  // Wrap in a list
-  BufferList serialized;
-  throw_if_not_ok(serialized.add_buffer(std::move(buf)));
+  // Wrap in a list. Pass the buffer's allocator to avoid copying with the
+  // default allocator.
+  BufferList serialized{buff.get_allocator()};
```

Comment:
This looks weird, getting the allocator from another object. Can't we get it from another source?

Reply:
We can use `serialization_buffer_resource()` but I didn't want to duplicate the call in the function, and we need to make sure the same allocator is used in both cases.

I can create the list first and pass the list's allocator to the buffer, to make the parent-child relationship clearer.

Reply:
That would be better.

Reply:
Oh we do this in every method of `Curl`. I'd prefer we don't change the code structure; I plan to remove the intermediary `BufferList` in the future either way.